tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/annotations/annotation.bal

```diff
@@ -27,6 +27,48 @@ function testAnnotOnBoundMethod() {
     assertEquality(102, rec.i);
 }
 
+type AnnotationRecord record {|
+    int minValue;
+    int maxValue;
+|};
+
+annotation AnnotationRecord BarAnnotation on type ;
+annotation AnnotationRecord BarAnnotation2 on type ;
+
+@BarAnnotation {
+    minValue: 18,
+    maxValue: 36
+}
+type Bar int|float;
+
+function testAnnotationWithReferredType() {
+    typedesc<any> 'type = typeof Bar;
+    AnnotationRecord? 'annotation = 'type.@BarAnnotation;
+    if 'annotation is AnnotationRecord {
+        assertEquality('annotation.minValue, 18);
+        assertEquality('annotation.minValue, 36);
+    }
+
+    AnnotationRecord? 'annotation2 = 'type.@BarAnnotation2;
+    assertEquality('annotation2, ());
+}
+
+@BarAnnotation2 {
+    minValue: 18,
+    maxValue: 36
+}
+type Bar2 Bar3;
+type Bar3 int|float;
+
+function testAnnotationWithReferredType2() {
```

Comment:
Do these tests actually test your fix? They seem to pass for me on 2201.2.1 also.

Reply:
This tests is to ensure hat annotation details can be access.
The actual fix is regarding type informations. When we pass `Bar` as a function parameter it will passed as `int|float` in previous. But with this PR it will pass as `Bar`.
So the relevant tests for that are in `dependently_typed_functions.bal`

Reply:
Isn't a more accurate test something like

```ballerina
function testAnnotsOnTypeRefTypes() {
    var f1 = function (typedesc td) {
        AnnotationRecord? 'annotation = td.@BarAnnotation2;
        assertTrue('annotation is AnnotationRecord);
        AnnotationRecord rec = <AnnotationRecord> 'annotation;
        assertEquality(rec.minValue, 18);
        assertEquality(rec.minValue, 36);
    };
    f1(Bar2);

    var f2 = function (typedesc td) {
        AnnotationRecord? 'annotation = td.@BarAnnotation2;
        assertTrue('annotation is ());
    };    
    f2(Bar3);
}
```